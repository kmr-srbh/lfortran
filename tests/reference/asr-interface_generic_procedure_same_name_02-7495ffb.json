{
    "basename": "asr-interface_generic_procedure_same_name_02-7495ffb",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/interface_generic_procedure_same_name_02.f90",
    "infile_hash": "30c0caa6d248877158816c85b167c850cb4456b3e814eff74bca7c8d",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-interface_generic_procedure_same_name_02-7495ffb.stdout",
    "stdout_hash": "f685f926d8a13fda4e0ddc5d7c17f0ef24cce0827e6e2a37908dae34",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}