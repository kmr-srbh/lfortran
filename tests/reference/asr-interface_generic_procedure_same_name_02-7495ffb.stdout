(TranslationUnit
    (SymbolTable
        1
        {
            interface_generic_procedure_same_name_02:
                (Module
                    (SymbolTable
                        2
                        {
                            first_interface:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            i:
                                                (Variable
                                                    5
                                                    i
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    first_interface
                                    (FunctionType
                                        [(Integer 4)]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .true.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 5 i)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            first_interface_2:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            i:
                                                (Variable
                                                    6
                                                    i
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    first_interface_2
                                    (FunctionType
                                        [(Integer 4)]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .true.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 6 i)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            first_interface~genericprocedure:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            
                                        })
                                    first_interface~genericprocedure
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Interface
                                        ()
                                        .false.
                                        .false.
                                        .true.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            second_interface:
                                (GenericProcedure
                                    2
                                    second_interface
                                    [2 second_interface~genericprocedure
                                    2 second_interface_2]
                                    Public
                                ),
                            second_interface_2:
                                (Function
                                    (SymbolTable
                                        7
                                        {
                                            i:
                                                (Variable
                                                    7
                                                    i
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                ),
                                            second_interface_2:
                                                (Variable
                                                    7
                                                    second_interface_2
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    second_interface_2
                                    (FunctionType
                                        [(Integer 4)]
                                        (Integer 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 7 i)]
                                    []
                                    (Var 7 second_interface_2)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            second_interface~genericprocedure:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            second_interface:
                                                (Variable
                                                    4
                                                    second_interface
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                ),
                                            x:
                                                (Variable
                                                    4
                                                    x
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    second_interface~genericprocedure
                                    (FunctionType
                                        [(Integer 4)]
                                        (Integer 4)
                                        Source
                                        Interface
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 4 x)]
                                    []
                                    (Var 4 second_interface)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            third_interface:
                                (GenericProcedure
                                    2
                                    third_interface
                                    [2 third_interface~genericprocedure]
                                    Public
                                ),
                            third_interface~genericprocedure:
                                (Function
                                    (SymbolTable
                                        8
                                        {
                                            i:
                                                (Variable
                                                    8
                                                    i
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    third_interface~genericprocedure
                                    (FunctionType
                                        [(Integer 4)]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 8 i)]
                                    [(Assignment
                                        (Var 8 i)
                                        (IntegerConstant 1 (Integer 4) Decimal)
                                        ()
                                        .false.
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            ~first_interface:
                                (GenericProcedure
                                    2
                                    ~first_interface
                                    [2 first_interface~genericprocedure
                                    2 first_interface_2]
                                    Public
                                )
                        })
                    interface_generic_procedure_same_name_02
                    []
                    .false.
                    .false.
                )
        })
    []
)
